zookeeper/conf/zoo.cfg

```diff
@@ -45,9 +45,8 @@ clientPortAddress={{ bind_address }}
 
 {%- else %}
 
-{%- for garbage_string in zookeepers %}
-  {%- set myid, mynodename = garbage_string.split('+') %}
-server.{{ myid }}={{ mynodename }}:{{ quorum_port }}:{{ election_port }}
+{%- for node in zookeepers %}
+server.{{ node.id }}={{ node.address | safe }}:{{ quorum_port }}:{{ election_port }}
```

Comment:
I'm wandering, why do you use the `safe()` filter here? It appears that the node address could be hostname, FQDN or IP address. Any other value will not work in any case.